### PR TITLE
[JBPM-9461] Add a flag/switch to include or exclude process variables…

### PR DESCRIPTION
**Pull request by elguardian**
Jira : https://issues.redhat.com/browse/JBPM-9461

### CODE REVIEW

**Review by afalhambra — APPROVED** [review]
A very minor comments, otherwise it's looking great.

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java [hunk 1]
```diff
@@ -220,6 +222,22 @@ public void testQueryLikeToOperator() {
...
 
+    @Test
+    public void testQueryProcVarsExclusion() {
+
+        List<QueryParam> variables = list(likeTo("var_a", false, "a%"));
+        List<QueryParam> attributes = list(exclude(PROCESS_COLLECTION_VARIABLES));
+
+        List<ProcessInstanceWithVarsDesc> data = advanceVariableDataService.queryProcessByVariables(attributes, variables, queryContext);
+        if (queryContext.getCount() > 0) {
+            assertThat(data.size(), is(queryContext.getCount()));
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java:233]: done

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java [hunk 2]
```diff
@@ -220,6 +222,22 @@ public void testQueryLikeToOperator() {
...
+        }
+
+        for (ProcessInstanceWithVarsDesc p : data) {
+            Assert.assertEquals(0, p.getVariables().size());
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java:237]: done

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java [hunk 3]
```diff
@@ -300,6 +318,21 @@ public void testQueryTaskByVariables() {
...
+        List<QueryParam> attributes = list(equalsTo(PROCESS_ATTR_DEPLOYMENT_ID, "org.jbpm.test:test-module:1.0.0"), exclude(PROCESS_COLLECTION_VARIABLES));
+
+        List<String> potOwners = emptyList();
+        List<UserTaskInstanceWithPotOwnerDesc> data = advanceVariableDataService.queryUserTasksByVariables(attributes, variables, emptyList(), potOwners, queryContext);
+        if (queryContext.getCount() > 0) {
+            assertThat(data.size(), is(queryContext.getCount()));
+        }
+        for (UserTaskInstanceWithPotOwnerDesc p : data) {
+            Assert.assertEquals(0, p.getProcessVariables().size());
+        }
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java:333]: done

**Review by gmunozfe — APPROVED** [review]
Looks good to me, nailed down @elguardian !
Just in case you want to add one more test for completeness invoking the API _queryProcessByVariablesAndTask_ with attributes list that _exclude(PROCESS_COLLECTION_VARIABLES)_

**Comment by elguardian** [comment]
Jenkins retest this

**Pull request merged**